src/rulesets/reader.ts

```diff
@@ -1,53 +1,47 @@
-import { merge } from 'lodash';
+import { parse } from '@stoplight/yaml';
 import { readParsable } from '../fs/reader';
 import { RuleCollection } from '../types';
-import { IRulesetFile } from '../types/ruleset';
-import { formatAjv } from './ajv';
-import { resolvePath } from './path';
-import { validateRuleset } from './validation';
-
-export async function readRulesFromRulesets(...files: string[]): Promise<RuleCollection> {
-  const rulesets = await Promise.all(files.map(file => readRulesFromRuleset(file)));
-  return merge({}, ...rulesets);
-}
+import { IRuleset, IRulesetFile } from '../types/ruleset';
+import { mergeRulesets } from './merger';
+import { resolvePath } from './resolver';
+import { assertValidRuleset } from './validation';
+
+export async function readRulesFromRulesets(...uris: string[]): Promise<RuleCollection> {
+  const base: IRuleset = {
+    rules: {},
+  };
 
-export type Logger = {
-  log: (message?: string | undefined, ...args: any[]) => void;
-  error: (
-    input: string | Error,
-    options?:
-      | {
-          code?: string | undefined;
-          exit?: number | undefined;
-        }
-      | undefined,
-  ) => never;
-};
-
-async function readRulesFromRuleset(file: string): Promise<RuleCollection> {
-  const parsed = await readParsable(file, 'utf8');
-  const { data: ruleset } = parsed;
-  const errors = validateRuleset(ruleset);
-
-  if (errors.length) {
-    throw {
-      messages: [formatAjv(errors), `Provided ruleset '${file}' is not valid`],
-    };
+  for (const uri of uris) {
+    mergeRulesets(base, await readRulesFromRuleset('', uri));
   }
 
-  const extendz = (ruleset as IRulesetFile).extends;
-  let extendedRules = {};
-  if (extendz && extendz.length) {
-    extendedRules = await blendRuleCollections(
-      extendz.map(extend => {
-        return readRulesFromRuleset(resolvePath(file, extend));
-      }),
-    );
+  for (const [name, rule] of Object.entries(base.rules)) {
+    if (rule.severity === 'off') {
+      delete base.rules[name];
```

Comment:
Any particular reason to delete it rather than leave it in with severity off?

Reply:
`DiagnosticSeveirty` enum does not contain `off` member. :thinking: 
As a part of the merging process, I convert human-readable severity to DiagnosticSeverity equivalent. Shall we include it?

Reply:
Frankly, I'd rather not, but we could simply extend the built-in type we have here in Spectral.

Reply:
Deleting it will makes it more difficult to look at the final resolved object and understand what's going on, because things will literally be missing. So wouldn't be able to show a UI in studio for example, that lists all the rules and if they're on or off.

Reply:
Can we put a list of things like this into a new story for cleanup? This wouldn’t break BC in any meaningful way so it wouldn’t matter. Gotta tag today realistically

Reply:
Fixed in https://github.com/stoplightio/spectral/pull/232/commits/5eee301604fae36fd06ffad4e9b17c5a0dcb8095